### PR TITLE
Update unit tests to use @Test attribute

### LINKED ISSUE #340: Update unit tests to use @Test attribute, descriptive names
**Is your feature request related to a problem? Please describe.**
Many of the unit tests in the Q# standard and domain-specific libraries predate the existence of the new [`@Test()` attribute](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.diagnostics.test), and should be updated. The legacy method of discovering tests (namely, functions and operations in allow-listed namespaces whose names end in "Test") is more difficult to use with functionality such as test filtering and reporting, and doesn't allow for running tests on a variety of simulator targets.

For example, the [existing `ReverseTest` function](https://github.com/microsoft/QuantumLibraries/blob/main/Standard/tests/ArrayTests.qs#L120) is written as:

```Q#
function ReverseTest () : Unit {
    let array = [1, 2, 3];
    Ignore(Mapped(EqualityFactI(_, _, $"Reversed failed."), Zipped([3, 2, 1], Reversed(array))));
}
```

Using the new `@Test` attribute would allow for giving this test a more descriptive name that helps when reading test failure logs:

```
open Microsoft.Quantum.Diagnostics;

@Test("QuantumSimulator")
function ReversedIsCorrect() : Unit {
    AllEqualityFactI(Reversed([1, 2, 3]), [3, 2, 1], "Reversed([1, 2, 3]) was incorrect.");
}
```

### PR DESCRIPTION
**Pull request by ihristova11**
- Used the `@Test()` attribute where applicable.
- Flipped _actual_ and _expected_ in ` Standard/tests/Math/OperatorFunctionTests.qs`

Resolves #340

### CODE REVIEW

File: Standard/tests/Math/OperatorFunctionTests.qs [hunk 1]
```diff
@@ -17,64 +17,67 @@ namespace Microsoft.Quantum.Tests {
...
+    function NegationIsCorrect() : Unit {
+        EqualityFactI(NegationI(42), -42, "NegationI returned wrong output.");
+        NearEqualityFactD(NegationD(42.0), -42.0);
+        EqualityFactL(NegationL(42L), -42L, "NegationI returned wrong output.");
+        NearEqualityFactC(NegationC(Complex(-1.0, -2.0)), Complex(1.0, 2.0));
         NearEqualityFactCP(
-            ComplexPolar(1.0, PI() / 4.0),
-            NegationCP(ComplexPolar(1.0, 5.0 * PI() / 4.0))
+            NegationCP(ComplexPolar(1.0, 5.0 * PI() / 4.0)),
+            ComplexPolar(1.0, PI() / 4.0)
```
Comment (cgranade) [inline Standard/tests/Math/OperatorFunctionTests.qs:28]: [nit] minor whitespace issue here, it looks like?
```suggestion
            ComplexPolar(1.0, PI() / 4.0)
```

**Review by cgranade — APPROVED** [review]
Looks good, thank you for your contribution! It looks like the CLA still needs signed, but other than that, I think this is good to bring in once CI completes. Thank you again!

File: Standard/tests/Math/OperatorFunctionTests.qs (hunk 1 above)
Comment (ihristova11) [reply Standard/tests/Math/OperatorFunctionTests.qs]: Just committed your suggestion, should I do `/azp run` again? I noticed that `microsoft.QuantumLibraries.e2e` check stays as _Expected_ for a while.

Comment (cgranade) [reply Standard/tests/Math/OperatorFunctionTests.qs]: Ah, thanks for catching that; I think I need to run that comment, so let me go on and do that.

**Comment by ihristova11** [comment]
@cgranade Thank you very much for your assistance with this. :) Is it possible to add the _hactoberfest-accepted_ label as well as I a trying to participate this year?

**Pull request merged**